### ISSUE TITLE
Add ttnn.sort basic implementation

### ISSUE THREAD

**Issue by aczajkowskiTT** [tenstorrent/tt-metal#19891]
Building on #19548 keep implementing ttnn.sort based on the definition found here:
https://pytorch.org/docs/stable/generated/torch.sort.html

In addition to returning the correct shape, the operation should now return the correct data along the given dimension.  Consider starting with a few small tensors in the examples from pytorch.org listed above, or an example below, and verify result when sorting on dim=0, dim=1, and dim=2.
```
x = torch.tensor([
    [[6, 3, 9, 1],
     [5, 2, 8, 4],
     [7, 0, 3, 2]],

    [[1, 7, 4, 6],
     [8, 5, 2, 3],
     [0, 9, 6, 1]]
])
```

Focus on a single core implementation, and avoid large tensors so that the working set fits in L1 cache.  After the trivial tensor sort operations are working correctly, try tile-sized tensor shapes.
